kernel/include/kernel_arch_interface.h

```diff
@@ -180,6 +180,17 @@ void arch_switch_to_main_thread(struct k_thread *main_thread, char *stack_ptr,
  * @retval -EINVAL If the floating point disabling could not be performed.
  */
 int arch_float_disable(struct k_thread *thread);
+
+/**
+ * @brief Enable floating point context preservation
+ *
+ * The function is used to enable the preservation of floating
+ * point context information for a particular thread.
+ *
```

Comment:
Add some text, also, here, that this API is optional , and depends on the architecture.

Reply:
I added such text.